master/internal/agent/agent.go

```diff
@@ -103,9 +103,19 @@ func (a *agent) Receive(ctx *actor.Context) error {
 		ctx.Respond(&proto.GetSlotsResponse{Slots: slots})
 	case *proto.EnableAgentRequest:
 		ctx.Tell(a.slots, patchSlot{Enabled: true})
+		ctx.Tell(a.resourcePool, sproto.EnableAgent{Agent: ctx.Self()})
 		ctx.Respond(&proto.EnableAgentResponse{Agent: a.summarize(ctx).ToProto()})
 	case *proto.DisableAgentRequest:
-		ctx.Tell(a.slots, patchSlot{Enabled: false})
+		// Mark current agent as disabled with RP.
+		ctx.Tell(a.resourcePool, sproto.DisableAgent{Agent: ctx.Self(), Drain: msg.Drain})
+		// Update individual slot state.
+		ctx.Tell(a.slots, patchSlot{Enabled: false, Drain: msg.Drain})
+		// Kill both slotted and zero-slot tasks, unless draining.
+		if !msg.Drain {
+			for cid := range a.containers {
+				ctx.Tell(ctx.Self(), sproto.KillTaskContainer{ContainerID: cid})
```

Comment:
Side note, I don't think the task actors as is are aware of these kills, so they will count to max restarts (for trials). Maybe drop a comment with a ticket?